src/main/java/com/hida/DatabaseManager.java

```diff
@@ -0,0 +1,1058 @@
+
+        databaseStatement.executeUpdate(sqlQuery);
+
+        databaseStatement.close();
+    }
+
+    /**
+     * Retrieves the default setting stored at the column in Settings table
+     *
+     * @param column name of the parameter to be changed
+     * @return setting stored at that location
+     * @throws SQLException thrown whenever there is an error with the database.
+     */
+    public Object retrieveSetting(String column) throws SQLException {
+        System.out.println("in retrieve settings");
+        Statement databaseStatement = DatabaseConnection.createStatement();
+
+        String sqlQuery = String.format("SELECT %s FROM %s;",
+                column, SETTINGS_TABLE);
+
+        Object parameter = null;
+        ResultSet result = databaseStatement.executeQuery(sqlQuery);
+
+        if (result.next()) {
+            parameter = result.getObject(column);
+        }
+
+        databaseStatement.close();
+        return parameter;
+    }
+
+    /**
+     * Adds a requested amount of formatted ids to the database.
+     *
+     * @param list list of ids to check.
+     * @param amountCreated Holds the true size of the list as list.size method
+     * can only return the maximum possible value of an integer.
+     * @param prefix The string that will be at the front of every id.
+     * @param tokenType Designates what characters are contained in the id's
+     * root.
+     * @param sansVowel Designates whether or not the id's root contains vowels.
+     * @param rootLength Designates the length of the id's root.
+     * @throws SQLException thrown whenever there is an error with the database
+     * @throws BadParameterException thrown whenever a malformed or invalid
+     * parameter is passed
+     */
+    public void addIdList(Set<Id> list, long amountCreated, String prefix, TokenType tokenType,
+            boolean sansVowel, int rootLength) throws SQLException, BadParameterException {
+        System.out.println("adding ids...");
+        Logger.info("Database being updated");
+        String valueQuery = "";
+        String insertQuery = "INSERT INTO " + ID_TABLE + "('" + ID_COLUMN + "') VALUES ";
+        Logger.info("ID inserted into: " + ID_TABLE + ", Column: " + ID_COLUMN);
+        int counter = 1;
+        Iterator<Id> listIterator = list.iterator();
+        while (listIterator.hasNext()) {
+            Id id = listIterator.next();
+
+            // concatenate valueQuery with the current id as a value to the statement
+            if (counter == 1) {
+                valueQuery += String.format(" ('%s')", id);
+            } else {
+                valueQuery += String.format(", ('%s')", id);
+            }
+
+            /* 500 is used because, to my knowledge, the max number of values that 
+             can be inserted at once is 500. The condition is also met whenever the
+             id is the last id in the list.            
+             */
+            if (counter == 500 || !listIterator.hasNext()) {

```

Comment:
This is probably not true (at least in general).  In general it will be limited by some other thing, like `max_allowed_packet` from MySQL.